### PR TITLE
Undo Push Notifications's project revert

### PR DESCRIPTION
**Pull request by daividssilverio**
## What's this?
This re-introduces the Push notifications project changes.

### Relationships
Re introduces #5225 

## Why do we want this?
Because we reverted the project while trying to find the source of the damned CALI bug.
Now it's time to re-revert it.

## How is it done?
```
git revert 02760c9534cbb5f844c49e63d0004b5f89b586fb
```
And bunch of manual conflict resolution.

### Why not another way?
Feel free to make suggestion on how to proceed with this.

### Side effects
I'm not sure, I need more people to test this and make sure everything is still working.

## Review hints
Please, check the files that had conflicts and to be manually solved:
- [x] both modified:   Toggl.Core.Tests.Sync/Helpers/AppServices.cs
- [x] both modified:   Toggl.Core.Tests/TestDependencyContainer.cs
- [x] both modified:   Toggl.Core.Tests/UI/ViewModels/BaseViewModelTests.cs
- [x] both modified:   Toggl.Core.Tests/UI/ViewModels/MainTabBarViewModelTests.cs
- [x] both modified:   Toggl.Core.Tests/UI/ViewModels/MainViewModelTests.cs
- [x] both modified:   Toggl.Core.UI/Navigation/ViewModelLoader.cs
- [x] both modified:   Toggl.Core.UI/ViewModels/MainTabBarViewModel.cs
- [x] both modified:   Toggl.Core.UI/ViewModels/MainViewModel.cs
- [x] both modified:   Toggl.Core/Analytics/BaseAnalyticsService.cs
- [x] both modified:   Toggl.Core/Analytics/IAnalyticsService.cs
- [x] both modified:   Toggl.Core/DependencyContainer.cs
- [x] deleted by us:   Toggl.Core/Diagnostics/MeasuredOperation.cs
- [x] both modified:   Toggl.Core/Interactors/IInteractorFactory.cs
- [x] both modified:   Toggl.Core/Services/BackgroundService.cs
- [x] both modified:   Toggl.Core/TogglSyncManagerFactory.cs
- [x] both modified:   Toggl.Droid/Fragments/MainFragment.cs
- [x] both modified:   Toggl.Droid/Resources/xml/RemoteConfigDefaults.xml
- [x] both modified:   Toggl.Droid/Startup/AndroidDependencyContainer.cs
- [x] both modified:   Toggl.Droid/Startup/TogglApplication.cs
- [x] both modified:   Toggl.Droid/Toggl.Droid.csproj
- [x] both modified:   Toggl.Droid/ViewHelpers/TimeEntryViewData.cs
- [x] both modified:   Toggl.Droid/packages.config
- [x] both modified:   Toggl.iOS/Startup/AppDelegate.cs
- [x] both modified:   Toggl.iOS/Startup/IosDependencyContainer.cs
- [x] both modified:   Toggl.iOS/Toggl.iOS.csproj
- [x] both modified:   Toggl.iOS/packages.config

Also, the old `RemoteConfigServiceIos` and `RemoteConfigServiceAndroid` are 🔥 because they are not being used anymore.

## Tests

Integration tests: https://app.bitrise.io/build/1f40f930127d8906 - do not merge this branch unless these tests are passing.

Syncing tests: https://app.bitrise.io/build/b90856200648d60c - do not merge this branch unless these tests are passing.

Detail: #5614 and #5572 have been cherry-picked to stay in place due to uncertainty whether or not the related tests should be there or not. They can be reintroduced when the push notifications api is available and released for everybody in production.

## :squid: Permissions
I'd like at least someone from @toggl/androids and someone from @toggl/ios-senseis and the @heytherewill (architect) to review this.
Merge it, if you know what you are doing. Please check if with the current release managers before doing that though.

### CODE REVIEW

File: Toggl.Droid/Resources/xml/RemoteConfigDefaults.xml [hunk 1]
```diff
@@ -7,5 +7,13 @@
     <entry>
         <key>UserRatingDialogTrigger</key>
         <value>none</value>
-    </entry> 
-</defaultsMap>
+    </entry>
+    <entry>
+        <key>register_push_notifications_token_with_server</key>
```
Comment (juanlaube) [inline Toggl.Droid/Resources/xml/RemoteConfigDefaults.xml]:
```suggestion
        <key>PushNotificationsRegisterTokenWithServer</key>
```

File: Toggl.Droid/Resources/xml/RemoteConfigDefaults.xml [hunk 2]
```diff
@@ -7,5 +7,13 @@
...
+        <value>false</value>
+    </entry>
+    <entry>
+        <key>handle_push_notifications</key>
```
Comment (juanlaube) [inline Toggl.Droid/Resources/xml/RemoteConfigDefaults.xml]:
```suggestion
        <key>PushNotificationsHandle</key>
```

File: Toggl.Core/Analytics/BaseAnalyticsService.cs [hunk 3]
```diff
@@ -264,6 +264,18 @@ protected BaseAnalyticsService()
         [AnalyticsEvent("SuggestionProvider")]
         public IAnalyticsEvent<SuggestionProviderType> SuggestionStarted { get; protected set; }
 
+        [AnalyticsEvent("NumberOfEntitiesFetched")]
+        public IAnalyticsEvent<string> PushInitiatedSyncFetch { get; protected set; }
```
[the remark this answers is not in the document]
Comment (daividssilverio) [reply Toggl.Core/Analytics/BaseAnalyticsService.cs]: Ok, I'll update the branch.

File: Toggl.Droid/Resources/xml/RemoteConfigDefaults.xml (hunk 2 above)
Comment (daividssilverio) [reply Toggl.Droid/Resources/xml/RemoteConfigDefaults.xml]: It's not a handle for push notifications, it's "handle push notifications", it's used like a boolean.

File: Toggl.Droid/Resources/xml/RemoteConfigDefaults.xml [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
-<defaultsMap>
-    <entry>
-        <key>UserRatingDialogDelayInDays</key>
-        <value>5</value>
-    </entry>
-    <entry>
-        <key>UserRatingDialogTrigger</key>
-        <value>none</value>
-    </entry>
+<defaultsMap>
```
Comment (juanlaube) [inline Toggl.Droid/Resources/xml/RemoteConfigDefaults.xml]: 👀 that's a lot of whitespaces

File: Toggl.Core.Tests/Interactors/PushNotifications/UnsubscribeFromPushNotificationsInteractorTests.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        public UnsubscribeFromPushNotificationsInteractorTests()
+        {
+            interactor = new UnsubscribeFromPushNotificationsInteractor(
+                PushNotificationsTokenService,
+                pushNotificationsTokenStorage,
+                Api);
+        }
+
+        [Fact, LogIfTooSlow]
+        public async Task ClearsThepushNotificationsTokenStorage()
```
Comment (dshr) [inline Toggl.Core.Tests/Interactors/PushNotifications/UnsubscribeFromPushNotificationsInteractorTests.cs]: Perhaps:
```suggestion
        public async Task ClearsThePushNotificationsTokenStorage()
```
?

**Pull request merged**